File: admin/utils/vite.py
from functools import cache
import json

from pathlib import Path

MANIFEST_PATH = (
    Path(__file__).parent.parent.parent / "web/dist/static/.vite/manifest.json"
)


@cache
def generate_vite_tags() -> str:
    if not MANIFEST_PATH.exists():
        raise FileNotFoundError("manifest.json not found")

    manifest_json = json.loads(MANIFEST_PATH.read_text())

    tag = ""

    for style in manifest_json["index.html"]["css"]:
        tag += f'<link rel="stylesheet" crossorigin href="/{style}">'

    if manifest_json["index.html"]["file"]:
        tag += f'<script type="module" crossorigin src="/{manifest_json["index.html"]["file"]}"></script>'

    return tag.strip()

File: internal/client/vite/vite.go
package vite

import (
	"encoding/json"
	"log"
)

type manifest struct {
	IndexHTML struct {
		CSS     []string `json:"css"`
		File    string   `json:"file"`
		IsEntry bool     `json:"isEntry"`
		Src     string   `json:"src"`
	} `json:"index.html"`
}

func GenerateViteTags(manifestString string) string {
	var manifest manifest
	if err := json.Unmarshal([]byte(manifestString), &manifest); err != nil {
		log.Fatal(err)
	}

	var tags string

	csses := manifest.IndexHTML.CSS
	if len(csses) > 0 {
		for _, css := range csses {
			tags += "<link rel=\"stylesheet\" crossorigin href=\"/static/" + css + "\">"
		}
	}

	file := manifest.IndexHTML.File
	if file != "" {
		tags += "<script type=\"module\" crossorigin src=\"/static/" + file + "\"></script>"
	}

	return tags
}
